src/pages/en/reference/api-reference.md

```diff
@@ -701,6 +701,149 @@ Pagination will pass a `page` prop to every rendered page that represents a sing
+- `injectedFrontatter` - An object of frontmatter [injected via remark or rehype plugins.](/en/guides/markdown-content/#example-injecting-frontmatter) Set to type `any`.
+
+```astro
+---
+import { getEntry } from 'astro:content';
+const entry = await getEntry('blog', 'entry-1.md');
+
+const { Content, headings, injectedFrontmatter } = await entry.render();
+---
+```
+
+See [the rendering entry contents guide](/en/guides/content-collections/#rendering-entry-content) for complete usage examples.
+
+### `getCollection()`
+
+**Type:** `(collection: string, filter?: (entry: CollectionEntry<collection>) => boolean) => CollectionEntry<collection>[]`
+
+`getCollection()` is a function that retrieves a list of content collection entries by collection name.
+
+It returns all items in the collection by default, and accepts an optional `filter` function to narrow by entry properties. This allows you to query for only some items in a collection based on `id`, `slug`, or frontmatter values via the `data` object.
+
+```astro
+---
+import { getCollection } from 'astro:content';
+
+// Get all `src/content/blog/` entries
+const allBlogPosts = await getCollection('blog');
+
+// Only return posts with `draft: true` in the frontmatter
+const draftBlogPosts = await getCollection('blog', ({ data }) => {
+  return data.draft === true;
+});
+---
+```
+
+[See the `getCollection()` guide section](/en/guides/content-collections/#getcollection) for complete usage examples.
+
+### `getEntry()`
+
+**Type:** `(collection: string, id: string) => CollectionEntry<collection>`
+
+`getEntry()` is a function that retrieves a single collection entry by collection name and [entry `id`](#id).
+
+
+```astro
+---
+import { getEntry } from 'astro:content';
+
+const enterprise = await getEntry('blog', 'enterprise.md');
+---
+```
+
+### `defineCollection()`
+
+A utility to configure a collection in a `src/content/config.*` file. This is often used when configuring a collection schema.
+
+This function accepts the following properties:
+
+#### `schema`
+
+**Type:** `TSchema extends ZodAnyObject | undefined`
+
+`schema` is an optional object to configure the type and shape of document frontmatter for a collection. Each object value must use [a Zod validator](https://github.com/colinhacks/zod).
+
+#### `slug()`
+
+**Type:** `(entry: { defaultSlug: string } & Omit<CollectionEntry, 'slug'>) => string | undefined`
+
+`slug()` is an optional function to override the default entry slug generated by Astro. This function receives the `defaultSlug`, entry `id`, unparsed document `body`, and `data` parsed by your configured schema, if any.
+
+#### Usage
+
+Say you have a collection named `src/content/blog/`, where each Markdown entry has a title and optional `permalink`. You can use `defineCollection` to enforce this frontmatter with the `schema` property. Then, you can map the `permalink` to the entry's `slug` if it exists:
```

Comment:
```suggestion
```
Like all the other times a code sample is used on this page, I would tend not to give this any preamble or explain what it's doing. The guide page shows how to define a collection, enforce frontmatter properties and customize the slug. The code samples here can just be "what it looks like" without much commentary.

I do like the code comment for the export! I think that's sufficient for what needs to go here!